server/src/com/cloud/network/IpAddressManagerImpl.java

```diff
@@ -936,8 +942,8 @@ public PublicIp assignDedicateIpAddress(Account owner, final Long guestNtwkId, f
                 displayIp = vpc.isDisplay();
             }
 
-            return fetchNewPublicIp(dcId, null, null, owner, VlanType.VirtualNetwork, guestNtwkId, isSourceNat, false, null, false, vpcId, displayIp);
-
+            ip = fetchNewPublicIp(dcId, null, null, owner, VlanType.VirtualNetwork, guestNtwkId, isSourceNat, true, null, false, vpcId, displayIp);
```

Comment:
You can return directly the call to `fetchNewPublicIp`.

I have not checked. What is that boolean flag you changed there?

Reply:
@rafaelweingartner actually there is a `ip` variable that is used in `finally` which is why I assign and then return see the code below this line.

Reply:
ok, no I saw it.
Thanks